irods_specific_implementation/irods_rest_api_base.h

```diff
@@ -137,69 +138,103 @@ namespace irods::rest
         } // dtor
 
     protected:
-        auto authenticate(
-              const std::string& _user_name
-            , const std::string& _password
-            , const std::string& _auth_type) -> void
+        template <typename ...Args>
+        void critical(const std::string_view _fmt, Args&&... _args) const
+        {
+            logger_->critical(_fmt, std::forward<Args>(_args)...);
+        } // critical
+
+        template <typename ...Args>
+        void error(const std::string_view _fmt, Args&&... _args) const
+        {
+            logger_->error(_fmt, std::forward<Args>(_args)...);
+        } // error
+
+        template <typename ...Args>
+        void warn(const std::string_view _fmt, Args&&... _args) const
+        {
+            logger_->warn(_fmt, std::forward<Args>(_args)...);
+        } // warn
+
+        template <typename ...Args>
+        void info(const std::string_view _fmt, Args&&... _args) const
+        {
+            logger_->info(_fmt, std::forward<Args>(_args)...);
+        } // info
+
+        template <typename ...Args>
+        void debug(const std::string_view _fmt, Args&&... _args) const
+        {
+            logger_->debug(_fmt, std::forward<Args>(_args)...);
+        } // debug
+
+        template <typename ...Args>
+        void trace(const std::string_view _fmt, Args&&... _args) const
+        {
+            logger_->trace(_fmt, std::forward<Args>(_args)...);
+        } // trace
```

Comment:
consider flipping the order of these - so they match the 'higher' log level at the 'top'.

assuming that doesn't actually change any logic.

Reply:
done